pkg/cmd/pipeline/logs_test.go

```diff
@@ -239,270 +127,97 @@ func TestLogs_interactive_get_all_inputs(t *testing.T) {
-					tb.PipelineRunStatusCondition(apis.Condition{
-						Status: corev1.ConditionTrue,
-						Reason: resources.ReasonSucceeded,
-					}),
-					// pipeline run started 3 minutes ago
-					tb.PipelineRunStartTime(clock.Now().Add(-3*time.Minute)),
-					// takes 10 minutes to complete
-					cb.PipelineRunCompletionTime(clock.Now().Add(10*time.Minute)),
-				),
-			),
+		{
+			name:      "Specify not exist pipelinerun",
+			command:   []string{"logs", "pipeline", "pipelinerun", "-n", "ns"},
+			namespace: "ns",
+			input:     cs,
+			wantError: true,
+			want:      "pipelineruns.tekton.dev \"pipelinerun\" not found",
 		},
-		Namespaces: []*corev1.Namespace{
-			{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: ns,
-				},
-			},
+		{
+			name:      "Specify invalid limit number",
+			command:   []string{"logs", pipelineName, "-n", ns, "--limit", "-1"},
+			namespace: "",
+			input:     cs2,
+			wantError: true,
+			want:      "limit was -1 but must be a positive number",
 		},
-	})
-
-	tests := []promptTest{
 		{
-			name:    "basic interaction",
-			cmdArgs: []string{pipelineName},
-
-			procedure: func(c *goexpect.Console) error {
-				if _, err := c.ExpectString("output-pipeline"); err != nil {
-					return err
-				}
-
-				if _, err := c.SendLine(string(terminal.KeyEnter)); err != nil {
-					return err
-				}
-
-				if _, err := c.ExpectString("Select pipelinerun:"); err != nil {
-					return err
-				}
-
-				if _, err := c.SendLine(string(terminal.KeyArrowDown)); err != nil {
-					return err
-				}
-
-				if _, err := c.ExpectString(prName2 + " started 3 minutes ago"); err != nil {
-					return err
-				}
-
-				if _, err := c.SendLine(string(terminal.KeyArrowDown)); err != nil {
-					return err
-				}
-
-				if _, err := c.ExpectString(prName + " started 5 minutes ago"); err != nil {
-					return err
-				}
-
-				if _, err := c.SendLine(string(terminal.KeyEnter)); err != nil {
-					return err
-				}
-
-				return nil
-			},
+			name:      "Specify last flag",
+			command:   []string{"logs", pipelineName, "-n", ns, "-L"},
+			namespace: "default",
+			input:     cs3,
+			wantError: false,
+			want:      "",
 		},
 	}
-	opts := logOpts(prName, ns, 2, false, cs)
-	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			RunPromptTest(t, test, opts)
+
+	for _, tp := range testParams {
+		t.Run(tp.name, func(t *testing.T) {
+			p := &test.Params{Tekton: tp.input.Pipeline, Clock: clock, Kube: tp.input.Kube}
+			if tp.namespace != "" {
+				p.SetNamespace(tp.namespace)
+			}
+			c := Command(p)
+
+			out, err := test.ExecuteCommand(c, tp.command...)
+			if tp.wantError {
+				if err == nil {
+					t.Errorf("error expected here")
+				}
+				test.AssertOutput(t, tp.want, err.Error())
+			} else {
+				if err != nil {
+					t.Errorf("unexpected Error")
+				}
+				test.AssertOutput(t, tp.want, out)
+			}
 		})
 	}
 }
 
-func TestLogs_interactive_limit_1(t *testing.T) {
+func TestPipelineLog2(t *testing.T) {
```

Comment:
```suggestion
func TestPipelineLog_Interactive(t *testing.T) {
```

Let's have a more descriptive name. It looks like this series of tests is about testing the interactive aspects of pipeline logs. Feel free to suggest something more apt.

Reply:
Thanks!! I totally agree and modified the name :thumbsup: